plenum/server/catchup/catchup_rep_gatherer.py

```diff
@@ -81,6 +88,57 @@ def start(self, cons_proof: ConsistencyProof):
         timeout = self._catchup_timeout(len(reqs))
         self._timer.schedule(timeout, self._request_txns_if_needed)
 
+    def stop(self, last_3pc: Optional[Tuple[int, int]] = None):
```

Comment:
In order to avoid errors/dirty state, maybe we should not use this service between rounds of catchup, that is instantiate every time for a single catch-up of a single ledger?